addons/csw/functions/fnc_assemble_canPickupWeapon.sqf

```diff
@@ -15,12 +15,11 @@
  * Public: No
  */
 
-params ["_staticWeapon"];
+params ["_vehicle"];
 
 // Assembly mode: [0=disabled, 1=enabled, 2=enabled&unload, 3=default]
-private _assemblyMode = [false, true, true, GVAR(defaultAssemblyMode)] select (_staticWeapon getVariable [QGVAR(assemblyMode), 3]);
-private _notCrewed = (crew _staticWeapon) isEqualTo [];
-private _deadCrew = !(alive (gunner _staticWeapon)); // need to eject body???
+private _assemblyMode = [false, true, true, GVAR(defaultAssemblyMode)] select (_vehicle getVariable [QGVAR(assemblyMode), 3]);
+private _notCrewed = (crew _vehicle) isEqualTo [];
+private _deadCrew = !(alive (gunner _vehicle)); // need to eject body???
 
 _assemblyMode && {_notCrewed || _deadCrew}
```

Comment:
Currently if a CSW is destroyed, it still shows the ACE interaction to disassemble it. You can select the action, but it stops immediately, as the condition for the progressbar requires the CSW to be alive.

The suggestion below removes the action for destroyed CSWs.

```suggestion
_assemblyMode && {alive _vehicle} && {_notCrewed || _deadCrew}
```